src/ast/query.rs

```diff
@@ -92,6 +92,8 @@ impl fmt::Display for Query {
 pub enum SetExpr {
     /// Restricted SELECT .. FROM .. HAVING (no ORDER BY or set operations)
     Select(Box<Select>),
+    /// SELECT .. FROM .. STARTING WITH .. CONNECT BY
+    ConnectBy(ConnectBy),
```

Comment:
Your rationale from the PR description is quite well stated and understandable and I think would be a good addition to the code

```suggestion
    /// 
    /// Note: `ConnectBy` is variant of `SetExpr`, rather than `TableFactor` or a field on `Select` 
    /// because, despite being part of a `SELECT .. FROM` clause, `CONNECT BY` does not compose
    /// with other table factors (e.g. `PIVOT`) in a `FROM`, nor does it permit most options usually available
    ///  in a `SELECT` (e.g. adding a `WHERE` clause in a `SELECT .. CONNECT BY` yields a parse error 
    /// from Snowflake).
    ConnectBy(ConnectBy),
```